src/renameKeyWith.js

```diff
@@ -0,0 +1,30 @@
+import { curry, equals, when } from 'ramda';
+
+import renameKeysWith from './renameKeysWith';
+
+/**
+ * Creates a new object with the own properties of the provided object, but the
+ * key `key` renamed according to logic of renaming function.
+ *
+ * Keep in mind that in case the new key name already existed on the object,
+ * the behaviour is undefined and the result may vary between various JS engines!
+ *
+ * @func renameKeyWith
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/1.5.0|v1.5.0}
+ * @category Object
+ * @sig (a -> b) -> c -> {a: *} -> {b: *}
```

Comment:
I think this is most appropriate
```suggestion
 * @sig (k -> k) -> k -> {k: v} -> {k: v}
```

Reply:
Thanks, will do :)